### PR TITLE
add images index and change root to images index page

### LINKED ISSUE #3: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [ ] New images that are added show up on the homepage.
- [ ] These images are persisted if the browser is closed or even if the
  server is restarted.
- [ ] Images are not displayed wider than 400px.
- [ ] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by stanleyphu**
Closes #3

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: app/views/images/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+<%= link_to 'New Image', new_image_path %>
+
+<table>
+  <tr>
+    <th>Title</th>
+    <th>Image</th>
+    <th></th>
+  </tr>
+
+  <% @images.reverse.each do |image| %>
```
Comment (shiboying) [inline app/views/images/index.html.erb]: As we follow MVC framework, the idea of `View` is to: `display data that got from controller`. Therefore, we don't prefer manipulate data on the View level. For example, the `reverse` is to change the order of `@images`. So I would suggest you to do the reverse operation in controller and give a more descriptive name for the instance variable like `@images_in_reverse_order`.

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+  def test_index
+    image1 = Image.new(title: 'test pic 1', url: 'https://www.pictures.com/abc')
+    image2 = Image.new(title: 'test pic 2', url: 'https://www.pictures.com/def')
+    image3 = Image.new(title: 'test pic 3', url: 'https://www.mypictures.com/abc123')
+
+    assert_difference 'Image.count', 3 do
+      image1.save
+      image2.save
+      image3.save
+    end
```
Comment (shiboying) [inline test/controllers/images_controller_test.rb]: As we are testing the `index` action, we can just setup the data directly, so we can change it to `Image.create(...)` and we don't need to assert the difference here.

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -1,6 +1,28 @@
...
+
+    get images_path
+
+    assert_response :success
+    assert_select '.js-image-row:nth-of-type(1) td:nth-child(1)', image3.title
+    assert_select '.js-image-row:nth-of-type(1) td:nth-child(2) img[src=?]', image3.url
+    assert_select '.js-image-row:nth-of-type(2) td:nth-child(1)', image2.title
+    assert_select '.js-image-row:nth-of-type(2) td:nth-child(2) img[src=?]', image2.url
+    assert_select '.js-image-row:nth-of-type(3) td:nth-child(1)', image1.title
+    assert_select '.js-image-row:nth-of-type(3) td:nth-child(2) img[src=?]', image1.url
```
Comment (shiboying) [inline test/controllers/images_controller_test.rb]: It is great you got a way to get all the element using the selector! Here I also paste what we discovery to use assert_select to assert order:
```Ruby
    expected = [image2.url, image1.url]
    assert_select 'img' do |elements|
      assert_equal expected, elements.map { |e| e[:src] }
    end
```

File: app/views/images/show.html.erb [hunk 4]
```diff
@@ -1,8 +1,10 @@
 <p>
   <strong>Title:</strong>
-  <span class="js-title"><%= @image.title %></span>
+  <span class='js-title'><%= @image.title %></span>
```
Comment (shiboying) [inline app/views/images/show.html.erb]: This change should belong to the previous commit. As it is minor, you don't need to change the commit but just give you a heads up.

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,5 +1,3 @@
 class ApplicationController < ActionController::Base
```
Comment (shiboying) [inline app/controllers/application_controller.rb]: I believe you also need to delete the corresponding test.

Comment (stanleyphu) [reply app/controllers/application_controller.rb]: fixed

File: app/views/images/index.html.erb (hunk 1 above)
Comment (stanleyphu) [reply app/views/images/index.html.erb]: fixed

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (stanleyphu) [reply test/controllers/images_controller_test.rb]: fixed

**Pull request merged**